src/poetry/factory.py

```diff
@@ -178,23 +180,28 @@ def create_legacy_repository(
         )
 
     @classmethod
-    def create_pyproject_from_package(cls, package: ProjectPackage, path: Path) -> None:
+    def create_pyproject_from_package(
+        cls, package: Package, path: Path | None = None
+    ) -> TOMLDocument:
         import tomlkit
 
-        from poetry.layouts.layout import POETRY_DEFAULT
-
-        pyproject = tomlkit.loads(POETRY_DEFAULT)
-        content = pyproject["tool"]["poetry"]
+        pyproject = tomlkit.document()
+        pyproject["tool"] = tomlkit.table()
+        pyproject["tool"]._is_super_table = True
+        content = pyproject["tool"]["poetry"] = tomlkit.table()
 
         content["name"] = package.name
         content["version"] = package.version.text
         content["description"] = package.description
         content["authors"] = package.authors
+        content["license"] = package.license.id if package.license else ""
+        content["classifiers"] = package.classifiers
+        content["readme"] = [*package.readmes]
```

Comment:
What about `maintainers`, `homepage` and so on? Should we consider everything from the [docs](https://python-poetry.org/docs/master/pyproject/)?

Reply:
Probably should eh? Will work it in.